blog/2023-05-11-retained-mqtt-messages-filtered.md

```diff
@@ -0,0 +1,18 @@
+---
+author: Jan Bouwhuis
+authorURL: https://twitter.com/jbouwh
+title: Recent MQTT changes to improve overall performance
+---
+
+The way Home Assistant's MQTT integration processes MQTT messages and how subscribes and unsubscribes are performed has changed.
+
+### Updates were made in the way MQTT subscribes, unsubscribes are performed
+
+Subscribes and unsubscribes are not immediately sent to the broker anymore. Debouncer code was added that will delay subscribes to the broker, and bundle them when more are added within the debounce time. At the point where the actual (un)subscribes are done, the (un)subscribes are bundled in one call to the broker. These changes will speed up the MQTT integration and the startup of integrations that register a lot of MQTT entities, especially when shared topics are subscribed (e.g. to publish the availability). In the MQTT debug logging the `mid` number shows what (un)subscribe calls have been bundled in one call to the MQTT broker.
```

Comment:
The comma after 'subscribes to the broker' is not necessary.

Change 'shared topics are subscribed' to 'shared topics are subscribed **to**'

Change 'bundled in one call' to 'bundled into one call'